src/common/data-modeler-service/ApplicationActions.ts

```diff
@@ -27,17 +33,49 @@ export class ApplicationActions extends DataModelerActions {
       currentActiveAsset?.type === EntityType.Model &&
       currentActiveAsset?.id
     ) {
-      this.databaseActionQueue.updatePriority(
-        currentActiveAsset.id,
-        DatabaseActionQueuePriority.InactiveModelProfile
-      );
+      const columns =
+        this.dataModelerStateService
+          .getEntityStateService(EntityType.Model, StateType.Derived)
+          .getById(currentActiveAsset.id)
+          .profile?.map((column) => column.name) || [];
+
+      columns.forEach((column) => {
+        Object.values(MetadataPriority).forEach((priority) => {
+          this.databaseActionQueue.updatePriority(
+            currentActiveAsset.id + column + priority,
+            getProfilePriority(
+              DatabaseActionQueuePriority.InactiveModelProfile,
+              DatabaseProfilesFieldPriority.NonFocused,
+              ProfileMetadataPriorityMap[priority]
+            )
+          );
+        });
+      });
     }
     this.dataModelerStateService.dispatch("setActiveAsset", [
```

Comment:
Where are we setting all column profiles as active when new active asset is a model?